checkbox-provider-ce-oem/bin/cpufreq_governors.py

```diff
@@ -0,0 +1,850 @@
+        logging.info(
+            "Restoring original governor to %s", (self.original_governors[0])
+        )
+        self.set_governor(self.original_governors[0])
+
+
+def main():
+    """
+    Run CPU Scaling Test.
+
+    This function runs a CPU scaling test based on the provided command-line
+    arguments.
+
+    Command-line Arguments:
+        -q, --quiet: Suppresses output.
+        -c, --capabilities: Only outputs CPU capabilities.
+        -d, --debug: Turns on debug level output for extra information
+                     during the test run.
+        --resource-format: Prints the capabilities in Checkbox resource job
+                           format.
+        --userspace: Runs the Userspace Governor test.
+        --performance: Runs the Performance Governor test.
+        --powersave: Runs the Powersave Governor test.
+        --ondemand: Runs the Ondemand Governor test.
+        --conservative: Runs the Conservative Governor test.
+        --schedutil: Runs the Schedutil Governor test.
+
+    Returns:
+        int: The exit code of the test run. 0 if successful, 1 otherwise.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-q", "--quiet", action="store_true", help="Suppress output."
+    )
+    parser.add_argument(
+        "-c",
+        "--capabilities",
+        action="store_true",
+        help="Only output CPU capabilities.",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        action="store_true",
+        help="Turn on debug level output for extra info during test run.",
+    )
+    parser.add_argument(
+        "--resource-format",
+        action="store_true",
+        help="Print the capabilities in Checkbox resource job format.",
+    )
+    parser.add_argument(
+        "--userspace",
+        action="store_true",
+        help="Run Userspace Governor test",
+    )
+    parser.add_argument(
+        "--performance",
+        action="store_true",
+        help="Run Performance Governor test",
+    )
+    parser.add_argument(
+        "--powersave",
+        action="store_true",
+        help="Run Powersave Governor test",
+    )
+    parser.add_argument(
+        "--ondemand",
+        action="store_true",
+        help="Run Ondemand Governor test",
+    )
+    parser.add_argument(
+        "--conservative",
+        action="store_true",
+        help="Run Conservative Governor test",
+    )
+    parser.add_argument(
+        "--schedutil",
+        action="store_true",
+        help="Run Schedutil Governor test",
+    )
+    args = parser.parse_args()
+
+    logger = init_logger()
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+
+    if args.quiet or args.resource_format:
+        # Not logging anything
+        logger.setLevel(logging.CRITICAL + 1)
+
+    test = CPUScalingTest()
+    if not test.get_cpu_freq_directories():
+        logging.info("CPU Frequency Scaling not supported")
+        return 1
```

Comment:
Should we use the `exit_code` here?

Reply:
I assumed that If "CPU Frequency Scaling not supported" or if having a problem in get_cpu_freq_directories(), the whole CPU governor test should fail directly. The following tests don't need to be executed.  This is the reason to `return 1` directly here.